### PR TITLE
feat: add custom refs plugin rules: no-refs-siblings and no-circular-refs

### PR DESCRIPTION
**Pull request by tatomyr**
Created two rules in the scope of investigating issues https://github.com/Redocly/redocly-cli/issues/1329 and 
 https://github.com/Redocly/redoc/issues/2474. 
Also related: https://github.com/Redocly/redocly-cli/issues/644

### CODE REVIEW

File: custom-plugin-rules/refs-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Authors:
+
+- [`@tatomyr`](https://github.com/tatomyr) Andrew Tatomyr (Redocly)
+
+## What this does and why
+
+Dealing with $refs might be quite a challenge sometimes.
+This plugin introduces a few rules that could give insights into how the $refs are used.
+
+### No $refs siblings
```
Comment (lornajane) [inline custom-plugin-rules/refs-plugin/README.md:12]: Maybe no _unexpected_ $refs siblings? Because we do still allow the valid ones like `description`.

**Comment by lornajane** [comment]
I tried to test this and it didn't do what I expected. Did I find an edge case, or do I not understand what a circular reference is? My (I edited the `components/schema` section of the [Museum API example](https://github.com/Redocly/museum-openapi-example) example snippet is like this:

```
    TicketConfirmation:
      description: Unique confirmation code used to verify ticket purchase.
      $ref: '/#components/schemas/BuyMuseumTicketsResponse'
    BuyMuseumTicketsResponse:
      description: Details for a museum ticket after a successful purchase.
      type: object
      properties:
        confirmationCode: 
          $ref: '#/components/schemas/TicketConfirmation'
      required:
        - confirmationCode
```

**Comment by adamaltman** [comment]
(I converted it to a draft because I come to check this PR every week and forget it is still in progress. 🧠 )

**Comment by tatomyr** [comment]
Closing for this now.
